examples/chat/app.py

```diff
@@ -0,0 +1,157 @@
+import os
+import threading
+import time
+from pathlib import Path
+
+import streamlit as st
+import zmq
+from code_editor import code_editor
+from streamlit_autorefresh import st_autorefresh
+
+import vision_agent as va
+
+WORKSPACE = Path(os.environ.get("WORKSPACE", ""))
+ZMQ_PORT = os.environ.get("ZMQ_PORT", None)
+if ZMQ_PORT is None:
+    ZMQ_PORT = "5555"
+    os.environ["ZMQ_PORT"] = ZMQ_PORT
+
+
+CACHE = Path(".cache.pkl")
+SAVE = {
+    "name": "Save",
+    "feather": "Save",
+    "hasText": True,
+    "commands": ["save-state", ["response", "saved"]],
+    "response": "saved",
+    "style": {"bottom": "calc(50% - 4.25rem", "right": "0.4rem"},
+}
+agent = va.agent.VisionAgent(verbosity=1)
+
+st.set_page_config(layout="wide")
+
+if "file_path" not in st.session_state:
+    st.session_state.file_path = None
+
+if "messages" not in st.session_state:
+    st.session_state.messages = []
+
+if "updates" not in st.session_state:
+    st.session_state.updates = []
+
+if "input_text" not in st.session_state:
+    st.session_state.input_text = ""
+
+
+def update_messages(messages, lock):
+    new_chat = agent.chat_with_code(messages)
+    with lock:
+        for new_message in new_chat:
+            if new_message not in messages:
+                messages.append(new_message)
+
+
+def get_updates(updates, lock):
+    context = zmq.Context()
+    socket = context.socket(zmq.PULL)
+    socket.bind(f"tcp://*:{ZMQ_PORT}")
+
+    while True:
+        message = socket.recv_json()
+        with lock:
+            updates.append(message)
+        time.sleep(0.1)
+
+
+def submit():
+    st.session_state.input_text = st.session_state.widget
+    st.session_state.widget = ""
+
+
+update_lock = threading.Lock()
+message_lock = threading.Lock()
+
+st_autorefresh(interval=1000, key="refresh")
+
+
+def main():
+    st.title("Vision Agent")
```

Comment:
May be a description in streamlit UI or in the README indicating the media path has to be explicitly mentioned in the chat , else users will try to upload image using the upload utility and expect the conversational agent to know the path. Apart from that the streamlit app looks good

Reply:
good point let me add that